arrow-libs/core/arrow-meta/build.gradle

```diff
@@ -30,8 +30,7 @@ dependencies {
     testImplementation "io.kotlintest:kotlintest-runner-junit4:$KOTLIN_TEST_VERSION", excludeArrow
     def toolsJar = Jvm.current().getToolsJar()
 
-    if (!toolsJar)
-        throw new GradleException("tools.jar not found at your JAVA_HOME dir ${Jvm.current().getJavaHome().getAbsolutePath()}.\n" +
-                "Building with a JRE or JDK9 is currently not supported.")
-    testCompileOnly files(toolsJar)
+    if (toolsJar) {
+        testCompileOnly files(toolsJar)
```

Comment:
Meanwhile you could check if it's possible to remove this dependency for testing to have the same behavior with JDK8 or higher.

Reply:
Just tested `clean build` with JDK 8 and it doesn't seem to need it :) Removing it: a353f73

![](https://user-images.githubusercontent.com/260697/117144699-23b45b00-adaa-11eb-9faf-302a40cb6167.png)

Reply:
Great, thanks @pablisco !! :clap: I'd miss 2 tasks:

- Just JDK8 is checked here so I miss adding checks for other JDKs.
- Update the documentation.

I'll help you asap!!

Reply:
@rachelcarmena I've updated the documentation (with the JDK 16 caveat) and updated the build tests to test on multiple JDK versions